src/azure-cli/azure/cli/command_modules/acs/_helpers.py

```diff
@@ -22,6 +23,9 @@ def _populate_api_server_access_profile(api_server_authorized_ip_ranges, enable_
     else:
         authorized_ip_ranges = [ip.strip() for ip in api_server_authorized_ip_ranges.split(",")]
 
+    if profile.enable_private_cluster and authorized_ip_ranges:
+        raise CLIError('authorized_ip_ranges is not supported for private cluster')
```

Comment:
--authorized-ip-ranges is better. authorized_ip_ranges is internal name

Reply:
fixed in the new commit